src/test/java/core/basesyntax/service/impl/FileReaderServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.errors.ErrorMessages;
+import core.basesyntax.service.FileReaderService;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Testing FileReaderServiceImpl")
+class FileReaderServiceImplTest {
+    private static final String TEST_FILE_NAME = "testFilename.csv";
+    private static final String FAILED_TO_DELETE_TEST_FILE = "Failed to delete test file: ";
+    private FileReaderService fileReaderService;
+
+    @BeforeEach
+    void setUp() {
+        fileReaderService = new FileReaderServiceImpl();
+    }
+
+    @Test
+    void read_existFile_ok() {
+        createTestFile();
```

Comment:
this file is created only once in this method. But you delete the file after each test. Is this a sufficient approach?

Reply:
Ok, I'll improve it